messaging/src/main/java/org/axonframework/messaging/annotation/AnnotatedHandlerInspector.java

```diff
@@ -168,15 +222,38 @@ private void registerHandler(MessageHandlingMember<T> handler) {
         return AnnotatedHandlerInspector.createInspector(entityType,
                                                          parameterResolverFactory,
                                                          handlerDefinition,
-                                                         registry);
+                                                         registry,
+                                                         Collections.emptySet());
     }
 
     /**
      * Returns a list of detected members of the inspected entity that are capable of handling certain messages.
      *
      * @return a list of detected message handlers
+     * @deprecated use {@link #getAllHandlers()} or {@link #getHandlers(Class)} instead
      */
+    @Deprecated
     public List<MessageHandlingMember<? super T>> getHandlers() {
-        return handlers;
+        return getHandlers(inspectedType).collect(Collectors.toList());
+    }
+
+    /**
+     * Returns a list of detected members of given {@code type} that are capable of handling certain messages.
+     *
+     * @param type a type of inspected entity
+     * @return a stream of detected message handlers for given {@code type}
+     */
+    public Stream<MessageHandlingMember<? super T>> getHandlers(Class<?> type) {
+        return handlers.getOrDefault(type, new TreeSet<>(HandlerComparator.instance()))
```

Comment:
Looks like you're returning an empty TreeSet, but don't register that as the value of handlers (as would be the case in `computeIfAbsent`). If you just want to return an empty stream, it's better to use `getOrDefault(type, emptySet())`.

Reply:
`getOrDefault(type, emptySortedSet())` :)